src/include/network/postgres/portal.h

```diff
@@ -59,12 +59,20 @@ class Portal {
    * @return params for this query
    */
   common::ManagedPointer<const std::vector<parser::ConstantValueExpression>> Parameters() {
+    return common::ManagedPointer(reinterpret_cast<const std::vector<parser::ConstantValueExpression> *>(&params_));
+  }
+
+  /**
+   * @return modifiable params for this query (Some existing API requires this version. Use with caution.)
+   * TODO(lin): If we have time, we probably should refactor other APIs to always use the const version
```

Comment:
I don't know if this will realistically ever happen if we don't do it now -- do we have an idea of how bad the modification would be?

Reply:
As discussed over Slack, this would require casting a non-const managed pointer to a const one, which can be either hacky or verbose. We might eventually need the `ManagedPointer` class to support this behavior.